pkg/cmd/rollbackcopy/backuputils_test.go

```diff
@@ -0,0 +1,189 @@
+package rollbackcopy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+type dirEntry struct {
+	name      string
+	version   string
+	timestamp string
+}
+
+func createBackupDir(basedirpath string, dir *dirEntry) error {
+	fullpath := filepath.Join(basedirpath, dir.name)
+	if err := checkAndCreateDir(fullpath); err != nil {
+		return err
+	}
+	backupVersion := &backupVersion{ClusterVersion: dir.version, TimeStamp: dir.timestamp}
+	if err := putVersion(backupVersion, fullpath); err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkBackupDir(basedirpath string, dir *dirEntry) error {
+	fullpath := filepath.Join(basedirpath, dir.name)
+	if !dirExists(fullpath) {
+		return fmt.Errorf("Directory %s is not found.", fullpath)
+	}
+	backupVersion, err := getVersion(fullpath)
+	if err != nil {
+		return err
+	}
+	if backupVersion.ClusterVersion != dir.version || backupVersion.TimeStamp != dir.timestamp {
+		return fmt.Errorf("Directory mismatch: expected %v, but got %v", dir, backupVersion)
+	}
+	return nil
+}
+
+func Test_checkVersionsAndMoveDirs(t *testing.T) {
+	type args struct {
+		configDir        string
```

Comment:
value never changes between scenarios, so the field can be removed to visually declutter the test cases